internal/column-helpers/columnhelpers.go

```diff
@@ -0,0 +1,307 @@
+
+// Columns is the (untyped) interface used for ColumnHelpers
+type Columns interface {
+	// GetTextColumnsFormatter returns the default formatter for this columns instance
+	GetTextColumnsFormatter(options ...textcolumns.Option) TextColumnsFormatter
+
+	// GetColumnNamesAndDescription returns a map of column names to their respective descriptions
+	GetColumnNamesAndDescription() map[string]string
+
+	// GetDefaultColumns returns a list of columns that are visible by default
+	GetDefaultColumns() []string
+
+	// SetColumnFilters sets additional column filters that will be used whenever one of the other methods of this
+	// interface are called. This is for example used to filter columns with information on kubernetes in a non-k8s
+	// environment like local-gadget
+	SetColumnFilters(...columns.ColumnFilter)
+
+	// SetSorting sets what sorting should be applied when calling SortEntries() // TODO
+	SetSorting([]string) error
+
+	// SetFilters sets which filter to apply before emitting events downstream
+	SetFilters([]string) error
+
+	// EventHandlerFunc returns a function that accepts an instance of type *T and pushes it downstream after applying
+	// enrichers and filters
+	EventHandlerFunc(enrichers ...func(any)) any
+	EventHandlerFuncArray(enrichers ...func(any)) any
+	EventHandlerFuncSnapshot(key string, enrichers ...func(any)) any
+
+	// JSONHandlerFunc returns a function that accepts a JSON encoded event, unmarshal it into *T and pushes it
+	// downstream after applying enrichers and filters
+	JSONHandlerFunc(enrichers ...func(any)) func([]byte)
+	JSONHandlerFuncArray(enrichers ...func(any)) func([]byte)
+
+	// SetEventCallback sets the downstream callback
+	SetEventCallback(eventCallback any)
+	SetEventCallbackArray(eventCallback any)
+
+	// SetErrorCallback sets the function to use whenever errors occur in this library or are received in-band
+	SetErrorCallback(errorCallback ErrorCallback)
+
+	// EnableSnapshots initializes the snapshot collector, which is able to aggregate snapshots from several sources
+	// and can return (optionally cached) results on demand; used for top gadgets
+	EnableSnapshots(ctx context.Context, t time.Duration, ttl int)
+}
+
+type ColumnHelpers[T any] struct {
+	columns            *columns.Columns[T]
+	sortBy             []string
+	filters            []string
+	filterSpec         *filter.FilterSpec[T] // TODO: filter collection(!)
+	eventCallback      func(*T)
+	eventCallbackArray func([]*T)
+	errorCallback      ErrorCallback
+	snapshotCombiner   *snapshotcombiner.SnapshotCombiner[T]
+	enableSnapshots    bool
+	columnFilters      []columns.ColumnFilter
+}
+
+func NewColumnHelpers[T any](columns *columns.Columns[T]) Columns {
+	ch := &ColumnHelpers[T]{
+		columns: columns,
+	}
+	return ch
+}
+
+func (ch *ColumnHelpers[T]) EnableSnapshots(ctx context.Context, interval time.Duration, ttl int) {
+	if ch.eventCallbackArray == nil {
+		panic("EnableSnapshots needs EventCallbackArray set")
```

Comment:
This function is only used rarely and at very specific locations, so the idea behind using panic in those places is to avoid additional error handling (mostly 2-3 loc in go) and bubbling, which could also make the issue harder to spot and debug. I don't mind changing it to softer erroring, if that is the common preference, though.